frontend-desktop/src/main.rs

```diff
@@ -22,27 +38,38 @@ fn wsconnect() -> Client{
     // get a socket that is connected to the admin namespace
     let socket = ClientBuilder::new("http://localhost:4200")
         .namespace("/")
-        .on("message", receive_message)
+        .on("coffee", receive_message)
         .on("error", websocket_error)
         .connect()
         .expect("Connection failed");
     return socket
-}*/
-
-fn call(ui: MainWindow) {
-    ui.invoke_add_entry("Mustermann".into(), "wants to get a coffee".into());
 }
 
 fn main() {
-    let ui = MainWindow::new();
+    let ui: MainWindow = MainWindow::new();
+    let websocket_client = wsconnect();
+
+    let timer = Timer::default();
 
     let history_model = Rc::new(slint::VecModel::<HistoryItem>::from(vec![
         //HistoryItem { text: "wants to get a coffee".into(), user: "Foo".into() },
     ]));
 
+
     let main_window_weak = ui.as_weak();
+    let history_model_clone = history_model.clone();
+    timer.start(TimerMode::Repeated, std::time::Duration::from_millis(2), move || {
+        let mut guard: MutexGuard<'_, Option<JoinMessage>> = mutex.lock().unwrap();
+        if let Some(item) = &*guard {
+            let history_item: HistoryItem = HistoryItem { text: "wants to go for a coffee".into(), user: item.name.clone().into() };
+            history_model_clone.push(history_item);
```

Comment:
Why does this show up in the UI? I thought we were working on a clone.

Reply:
Thought so too but it seems to work. Need someone with more rust knowledge